### PR TITLE
Update clang-format to 10.0

### PR DESCRIPTION
**Pull request by DHowett**
This commit removes our local copy of clang-format 8 and replaces it
with a newly-built nuget package containing clang-format 10.

This resolves the inconsistency between our version of clang-format and
the one shipped in Visual Studio.

A couple minor format changes were either required or erroneously forced
upon us--chief among them is a redistribution of `*`s around SAL
annotations in inline class members of COM classes. Don't ask why; I
couldn't figure it out.

We had some aspirational goals for our formatting, which were left in
but commented out. Enabling them changes our format a little more than
I'm comfortable with, so I uncommented them and locked them to the
format style we've been using for the past year. We may not love it, but
our aspirations may not matter here any longer. Consistent formatting is
better than perfect formatting.

### CODE REVIEW

File: src/inc/til/static_map.h [hunk 1]
```diff
@@ -70,5 +70,5 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
     // 1. static_map's member types are all the same
     // 2. static_map's fourth template argument (otherwise undeduced) is how many pairs it contains
     template<typename First, typename... Rest>
-    static_map(First, Rest...)->static_map<std::conditional_t<std::conjunction_v<std::is_same<First, Rest>...>, typename First::first_type, void>, typename First::second_type, std::less<typename First::first_type>, 1 + sizeof...(Rest)>;
+    static_map(First, Rest...) -> static_map<std::conditional_t<std::conjunction_v<std::is_same<First, Rest>...>, typename First::first_type, void>, typename First::second_type, std::less<typename First::first_type>, 1 + sizeof...(Rest)>;
```
Comment (DHowett) [inline src/inc/til/static_map.h:73]: FORMAT: deduction guide gets a space around its arrow.

File: src/cascadia/PublicTerminalCore/HwndTerminal.cpp [hunk 2]
```diff
@@ -242,7 +242,7 @@ HRESULT HwndTerminal::Initialize()
     _terminal->Create(COORD{ 80, 25 }, 1000, *_renderer);
     _terminal->SetDefaultBackground(RGB(12, 12, 12));
     _terminal->SetDefaultForeground(RGB(204, 204, 204));
-    _terminal->SetWriteInputCallback([=](std::wstring & input) noexcept { _WriteTextToConnection(input); });
+    _terminal->SetWriteInputCallback([=](std::wstring& input) noexcept { _WriteTextToConnection(input); });
```
Comment (DHowett) [inline src/cascadia/PublicTerminalCore/HwndTerminal.cpp:245]: FORMAT: clang-format used to (weirdly) force the space before the `&`, now it forces no space

File: src/propsheet/PropSheetHandler.cpp [hunk 3]
```diff
@@ -34,7 +34,7 @@ class ConsolePropertySheetHandler WrlFinal : public RuntimeClass<RuntimeClassFla
     }
 
     // IPersist
-    STDMETHODIMP GetClassID(_Out_ CLSID* clsid) override
+    STDMETHODIMP GetClassID(_Out_ CLSID * clsid) override
```
Comment (DHowett) [inline src/propsheet/PropSheetHandler.cpp:37]: FORMAT: This whole file is filled with the issues I called out in the body.

Comment (skyline75489) [reply src/propsheet/PropSheetHandler.cpp:37]: There is some serious Objective-C vibe going around this one...

Comment (DHowett) [reply src/propsheet/PropSheetHandler.cpp:37]: Ah, objective-C, my other favorite language :)

File: tools/OpenConsole.psm1 [hunk 4]
```diff
@@ -351,9 +354,12 @@ function Invoke-ClangFormat {
 #.SYNOPSIS
 # runs code formatting on all c++ files
 function Invoke-CodeFormat() {
+    & "$env:OpenConsoleRoot\dep\nuget\nuget.exe" restore "$env:OpenConsoleRoot\tools\packages.config"
+    $clangPackage = ([xml](Get-Content "$env:OpenConsoleRoot\tools\packages.config")).packages.package | Where-Object id -like "clang-format*"
+    $clangFormatPath = "$env:OpenConsoleRoot\packages\$($clangPackage.id).$($clangPackage.version)\tools\clang-format.exe"
```
Comment (miniksa) [inline tools/OpenConsole.psm1:359]: Well that's awesome.

**Pull request merged**